### ISSUE TITLE
Unexpected packer compile error

### ISSUE THREAD

**Issue by ruanyouxing** [wbthomason/packer.nvim#1058]
- `nvim --version`: 0.7.2
- `git --version`: 2.37.2
- Operating system/version: Arch Linux
- Terminal name/version:Kitty 0.25.2

### Steps to reproduce
1.Run `:PackerSync` or anything that calls require('packer').compile()
2.All plugins cloned
3.Error triggered
### Actual behaviour
[ERROR] nvim/site/pack/packer/opt/packer.nvim/lua/packer/async.lua:20: Error in coroutine: .../site/pack/packer/opt/packer.nvim/lua/packer/compile.lua:674: attempt to index a nil value

`packer_compiled.lua`  couldn't create
### Expected behaviour
It runs normally and there's no error triggered
### packer files

<details>
<summary>Plugin specification file(s)</summary>

[pack.lua](https://github.com/ruanyouxing/nvim/blob/main/lua/core/pack.lua)

[modules folder](https://github.com/ruanyouxing/nvim/tree/main/lua/modules), based on [Cosynvim template](https://github.com/glepnir/cosynvim)
</details>

<details>
<summary>packer log file</summary>

Post the contents of ~/.cache/nvim/packer.nvim.log here
```
$ cat ~/.cache/nvim/packer.nvim.log`

`[ERROR Thu Sep  8 12:37:19 2022 3443993188514] ...im/site/pack/packer/opt/packer.nvim/lua/packer/async.lua:20: Error in coroutine: .../site/pack/packer/opt/packer.nvim/lua/packer/compile.lua:674: attempt to index a nil value`
```
</details>

<details>
<summary>packer compiled file</summary>

It couldn't create, so there's no `packer_compiled.lua`

</details>

**Comment by wbthomason** [comment]
Thanks for the report. Can you isolate a commit in `packer` or your config files, etc. after which the problem started? This will help us isolate the problem.

**Comment by ruanyouxing** [comment]
> Thanks for the report. Can you isolate a commit in `packer` or your config files, etc. after which the problem started? This will help us isolate the problem.

I'm not sure but I got the error from September 6th, maybe it's [50aeb90](https://github.com/wbthomason/packer.nvim/commit/50aeb9060cf64c3c27e6d7b11d7af9e209ed6c3b) or [6db20b4](https://github.com/wbthomason/packer.nvim/commit/6db20b4804b432beb04abe2a3e850e03e0ec1f27)

**Comment by wbthomason** [comment]
Did you make changes to your config around this time? The line in the error message has to do with resolving `wants` and `after` - those features haven't been touched in `packer` for a while, unless I've forgotten something.

**Comment by ruanyouxing** [comment]
Nope, I didn't change it at all, it ran properly in the past, when I use `after` for some plugins

**Comment by wbthomason** [comment]
Hmm. Might be a long-standing bug, then. If either of you can isolate a minimal reproducing example, that would be really helpful.

**Comment by ruanyouxing** [comment]
```lua
plugin ({'nvim-treesitter/nvim-treesitter',
  event = 'BufRead',
  run = ':TSUpdate',
  config = conf.treesitter,
})
plugin ({'nvim-treesitter/nvim-treesitter-refactor',
  after = 'nvim-treesitter',
})
plugin ({'JoosepAlviste/nvim-ts-context-commentstring',
  after = 'nvim-treesitter',
})
plugin ({'nvim-treesitter/nvim-treesitter-textobjects',
  after = 'nvim-treesitter',
})
plugin ({'windwp/nvim-ts-autotag', after = 'nvim-treesitter' })
plugin ({'p00f/nvim-ts-rainbow', after = 'nvim-treesitter' })
plugin ({'andymass/vim-matchup', after = 'nvim-treesitter' })
plugin ({'romgrk/nvim-treesitter-context',
  after = 'nvim-treesitter',
  config = conf.ts_context,
})
```
Well this is how I use `after` with treesitter plugins
And I asked myself why others' dotfiles work with `after`, except me

**Comment by ruanyouxing** [comment]
> > Did you make changes to your config around this time? The line in the error message has to do with resolving `wants` and `after` - those features haven't been touched in `packer` for a while, unless I've forgotten something.
> 
> That's gotta be it, coz I use `after` a lot,l and that happened occasionally while running `PackerSync`

@babygau oh i know the problem
Did you use `opt = true` and `after` at the same time?

**Comment by ruanyouxing** [comment]
@wbthomason I think i got the problem
maybe I use `after` and `opt = true` at the same time and it triggered errors

**Comment by ruanyouxing** [comment]
> @ruanyouxing No, I don't use these 2 keys at a time. This happen sometimes when I update the plugins, and also when I change the plugin with same name but different fork version.

@babygau what's your plugin can you show me?

**Comment by arminveres** [comment]
Hi there, has there been any progress on this? I am getting a similar error with neovim version 0.8 and the error:
```
[ERROR 18:41:51] async.lua:20: Error in coroutine: ...e/nvim/site/pack/packer/start/packer.nvim/lua/packer.lua:757: invalid value (boolean) at index 17 in table for 'concat'
```
